### PR TITLE
Implemented validations and Tests

### PR DESCRIPTION
**Pull request by GordonOtieno**
- Added the following validations:
    - For the User model:
      - Name must not be blank.
      - PostsCounter must be an integer greater than or equal to zero.
    - For the Post model:
       - Title must not be blank.
       - Title must not exceed 250 characters.
       - CommentsCounter must be an integer greater than or equal to zero.
       - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for User model and Post model' methods and validations.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -3,10 +3,13 @@ class Post < ApplicationRecord
   has_many :comments, foreign_key: :post_id
 
   belongs_to :author, class_name: 'User', foreign_key: 'author_id'
-  after_save :update_post_counter
+
+  validates :title, presence: true, length: { maximum: 100 }
```
Comment (DammyShittu) [inline app/models/post.rb]: - [ ] The maximum length should be 250, not 100. Kindly fix that.

File: spec/model/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  it 'likes_counter should be integer' do
+    subject.likes_counter = 'i'
+    expect(subject).to_not be_valid
+  end
+
+  it 'likes_counter should be greater than or equal to zero' do
+    subject.likes_counter = -1
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (DammyShittu) [inline spec/model/post_spec.rb]: - [ ] All the tests contained here are testing for negative scenarios. Kindly add one or more scenarios for when the Post is valid to account for all possible scenarios.

- [ ] Kindly add tests for the `Comment` and `Like` models.

File: spec/model/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  it 'postsCounter should be integer' do
+    subject.postsCounter = 'a'
+    expect(subject).to_not be_valid
+  end
+
+  it 'postsCounter should be greater than or equal to zero' do
+    subject.postsCounter = -1
+    expect(subject).to_not be_valid
+  end
```
Comment (DammyShittu) [inline spec/model/user_spec.rb:25]: - [ ] To cover all scenarios kindly add a test for when the User is actually valid.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @GordonOtieno,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] When I try to migrate your test database I get this error. You need to add test database configuration to your `config/database.yml` file, like this:

  ```
  test:
    <<: *default
    database: blog_app__ruby_test
  ```

  uripw

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/model/user_spec.rb (hunk 3 above)
Comment (GzimAsani) [reply spec/model/user_spec.rb:25]: - [ ] As asked by the previous reviewer kindly cover the scenario when the user is **valid**

File: spec/model/post_spec.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
```
Comment (GzimAsani) [inline spec/model/post_spec.rb]: - [ ] As asked by the previous reviewer kindly test the case when the post is also **valid**
- [ ] Also, please cover the scenarios when the like and comment is **valid**

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello  @GordonOtieno              :wave:   
 Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job adding like and comment tests :+1: 

 ## Required Changes 


- [ ] Check the comments below :arrow_double_down: 



 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @GordonOtieno       :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for making all the suggested changes good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**